exchanges/gateio/gateio_wrapper.go

```diff
@@ -1849,33 +1855,27 @@ func (g *Gateio) GetHistoricCandlesExtended(ctx context.Context, pair currency.P
 	if err != nil {
 		return nil, err
 	}
-	klineData := &kline.Item{
-		Interval: interval,
-		Asset:    a,
-		Pair:     formattedPair.Upper(),
-		Exchange: g.Name,
-	}
 	var dates *kline.IntervalRangeHolder
 	dates, err = kline.CalculateCandleDateRanges(start, end, interval, 0)
 	if err != nil {
 		return nil, err
 	}
-	var candlestickItems []kline.Candle
 	var fPair currency.PairFormat
 	fPair, err = g.GetPairFormat(a, true)
 	if err != nil {
 		return nil, err
 	}
+	candlestickItems := []kline.Candle{}
```

Comment:
There is a pre alloc trick that call Req.Size() so we can go `make([]kline.Candle, 0, req.Size())` and then append as necessary.